python/cudf_polars/cudf_polars/dsl/expressions/unary.py

```diff
@@ -233,6 +234,14 @@ def do_evaluate(
             else:
                 evaluated = self.children[1].evaluate(df, context=context)
                 arg = evaluated.obj_scalar if evaluated.is_scalar else evaluated.obj
+            if (
+                not POLARS_VERSION_LT_128
+                and isinstance(arg, plc.Scalar)
+                and dtypes.can_cast(column.obj.type(), arg.type())
+            ):  # pragma: no cover
+                arg = plc.unary.cast(
+                    plc.Column.from_scalar(arg, 1), column.obj.type()
+                ).to_scalar()
```

Comment:
Why is this `no cover`. Since we're testing with 1.29 in the test suite we should be able to write a test for it.